book/src/custom-log-output.md

```diff
@@ -0,0 +1,186 @@
+# Customizing the log output
+
+The way a printer outputs logs can be customized by providing a format string to `defmt`. The format string takes a set of metadata and format specifiers which can be used to include or exclude certain information when printing the logs.
+
+## Basics
+
+> The following log will be used as reference in the examples below: `defmt::debug!("hello");`
+
+The simplest format string is `"{s}"`. This prints the log and nothing else:
+
+```text
+hello
+```
+
+Arbitrary text can be added to the format string, which will be printed as specified with each log.
+For example, `"Log: {s}"`:
+
+```text
+Log: hello
+```
+
+Multiple specifiers can be included within a format string, in any order. For example `"[{L}] {s}"` prints:
+
+```text
+[DEBUG] hello
+```
+
+## Metadata specifiers
+
+There are several metadata specifiers available that can be used in a format string.
+
+#### Log - `{s}`
```

Comment:
```suggestion
### Log - `{s}`
```

I think a 3-# heading is correct here?

Reply:
I think I made these level-4 intentionally because it looks better in the book.